examples/src/test/scala/docs/cassandra/WordCountDocExample.scala

```diff
@@ -162,4 +168,198 @@ object WordCountDocExample {
+        repository: WordCountRepository) {
+      import WordCountProcessor._
+
+      // loading initial state from db
+      def init(): Behavior[Command] = {
+        Behaviors.withStash(10) { buffer =>
+          context.pipeToSelf(repository.loadAll(projectionId.id)) {
+            case Success(value) => InitialState(value)
+            case Failure(exc)   => throw exc
+          }
+
+          Behaviors.receiveMessage {
+            case InitialState(state) =>
+              context.log.debug("Initial state [{}]", state)
+              buffer.unstashAll(idle(state))
+            case other =>
+              context.log.debug("Stashed [{}]", other)
+              buffer.stash(other)
+              Behaviors.same
+          }
+        }
+      }
+
+      // waiting for next envelope
+      private def idle(state: Map[Word, Count]): Behavior[Command] =
+        Behaviors.receiveMessagePartial {
+          case Handle(envelope, replyTo) =>
+            val word = envelope.word
+            context.pipeToSelf(repository.save(projectionId.id, word, state.getOrElse(word, 0) + 1)) { saveResult =>
+              SaveCompleted(word, saveResult)
+            }
+            saving(state, replyTo) // will reply from SaveCompleted
+          case Stop =>
+            Behaviors.stopped
+          case _: InitialState =>
+            Behaviors.unhandled
+        }
+
+      // saving the new count for a word in db
+      private def saving(state: Map[Word, Count], replyTo: ActorRef[Try[Done]]): Behavior[Command] =
+        Behaviors.receiveMessagePartial {
+          case SaveCompleted(word, saveResult) =>
+            replyTo ! saveResult
+            saveResult match {
+              case Success(_)   => idle(state.updated(word, state.getOrElse(word, 0) + 1))
+              case Failure(exc) => throw exc // restart, reload state from db
+            }
+          case Stop =>
+            Behaviors.stopped
+        }
+    }
+    //#behaviorLoadingInitialState
+  }
+
+  object IllstrateActorLoadingStateOnDemand {
+    import akka.actor.typed.ActorRef
+    import akka.actor.typed.Behavior
+    import akka.actor.typed.SupervisorStrategy
+    import akka.actor.typed.scaladsl.ActorContext
+    import akka.actor.typed.scaladsl.Behaviors
+    import akka.projection.scaladsl.ActorHandler
+
+    class WordCountActorHandler(behavior: Behavior[WordCountProcessor.Command])(implicit system: ActorSystem[_])
+        extends ActorHandler[WordEnvelope, WordCountProcessor.Command](behavior) {
+      import akka.actor.typed.scaladsl.AskPattern._
+      import system.executionContext
+
+      private implicit val askTimeout: Timeout = 5.seconds
+
+      override def process(envelope: WordEnvelope, actor: ActorRef[WordCountProcessor.Command]): Future[Done] = {
+        actor.ask[Try[Done]](replyTo => WordCountProcessor.Handle(envelope, replyTo)).map {
+          case Success(_)   => Done
+          case Failure(exc) => throw exc
+        }
+      }
+    }
```

Comment:
This is the main API difference compared to #116
Here the users have to implement the ask themselves. Note that it requires ActorSystem in implicit scope.